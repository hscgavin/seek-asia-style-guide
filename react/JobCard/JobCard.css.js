import { saGrey2 } from '../../theme/palette/palette.js';

export default {
  '.root': {
    width: '100%',
    marginBottom: 0,
    borderStyle: 'solid',
    borderBottomWidth: '1px',
    borderColor: saGrey2
  },
  '.footer': {
    display: 'flex',
    justifyContent: 'space-between',
    paddingTop: 0
  },
  '.footer span': {
    verticalAlign: 'middle'
  },
  '.logo': {
    maxWidth: '100px'
  },
  '.company': {
    padding: 0
  },
  '.bodyDescription': {
    paddingTop: 0
  }
};
