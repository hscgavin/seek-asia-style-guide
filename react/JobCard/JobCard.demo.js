import React from 'react';
import PropTypes from 'prop-types';
import { JobCard, PageBlock } from 'seek-style-guide/react';

const JobCardContainer = ({ component: DemoComponent, componentProps }) => {
  return (
    <PageBlock style={{ width: '75%' }}>
      <DemoComponent {...componentProps} />
    </PageBlock>
  );
};

JobCardContainer.propTypes = {
  component: PropTypes.element,
  componentProps: PropTypes.object
};

export default {
  route: '/jobCard',
  title: 'Job Card',
  component: JobCard,
  container: JobCardContainer,
  initialProps: {
    job: {
      company: 'SEEK Asia',
      jobTitle: 'Developer',
      location: 'Kuala Lumpur',
      description: 'Responsibilities :Responsible for Client Relationship Management and Worker Performance Management. Responsible for full spectrum of human resource and admin function, include...',
      companyLogoUrl: 'https://siva.jsstatic.com/my/94463/images/logo/94463_logo_0_48885.png',
      salary: 'RM99999 - RM999999',
      postingDuration: '1h'
    }
  },
  options: [{
    label: 'Description',
    type: 'radio',
    states: [{
      label: 'No Description',
      transformProps: ({ ...props }) => ({
        ...props,
        job: {
          ...props.job,
          description: null
        }
      })
    }, {
      label: 'Description present',
      transformProps: ({ ...props }) => ({
        ...props,
        job: {
          ...props.job,
          description: 'Responsibilities :Responsible for Client Relationship Management and Worker Performance Management. Responsible for full spectrum of human resource and admin function, include...'
        }
      })
    }]
  },
  {
    label: 'Company Logo',
    type: 'radio',
    states: [{
      label: 'No Logo',
      transformProps: ({ ...props }) => ({
        ...props,
        job: {
          ...props.job,
          companyLogoUrl: null
        }
      })
    }, {
      label: 'Logo present',
      transformProps: ({ ...props }) => ({
        ...props,
        job: {
          ...props.job,
          companyLogoUrl: 'https://siva.jsstatic.com/my/94463/images/logo/94463_logo_0_48885.png'
        }
      })
    }]
  }]
};
