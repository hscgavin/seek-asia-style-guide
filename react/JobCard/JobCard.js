import React from 'react';
import PropTypes from 'prop-types';

import Text from '../Text/Text';
import Card from '../Card/Card';
import Section from '../Section/Section';
import LocationIcon from '../LocationIcon/LocationIcon';
import MoneyIcon from '../MoneyIcon/MoneyIcon';
import TimeIcon from '../TimeIcon/TimeIcon';
import styles from './JobCard.css';

const JobCard = ({ job }) => {
  return (
    <Card className={styles.root}>
      <Section>
        <Text intimate className={styles.company}>{job.company}</Text>
        <Text yelling>{job.jobTitle}</Text>
      </Section>
      { job.description && (
        <Section className={styles.bodyDescription}>
          <Text intimate>{job.description}</Text>
        </Section>
      )}
      <Section className={styles.footer}>
        <div>
          <Text intimate><LocationIcon /> {job.location}</Text>
          <Text intimate><MoneyIcon /> {job.salary}</Text>
          <Text intimate><TimeIcon /> {job.postingDuration}</Text>
        </div>
        { job.companyLogoUrl && (
        <div className={styles.companyLogoWrapper}>
          <img className={styles.companyLogo} src={job.companyLogoUrl} />
        </div>
        )}
      </Section>
    </Card>
  );
};

export default JobCard;
JobCard.propTypes = {
  job: PropTypes.shape({
    jobTitle: PropTypes.string.isRequired,
    company: PropTypes.string.isRequired,
    companyLogoUrl: PropTypes.string,
    description: PropTypes.string,
    location: PropTypes.string.isRequired,
    postingDuration: PropTypes.string.isRequired
  }).isRequired
};
