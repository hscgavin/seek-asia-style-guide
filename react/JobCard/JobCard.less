@import (reference) "~seek-asia-style-guide/theme";
@job-card-section-space: 0;
@job-card-label-indent: 5px;
@job-card-list-indent: 14px;
@job-card-text-space: 0;
@job-card-bottom-text-space: 3px;
@job-card-logo-indent: 15px;

.root {
  width: 100%;
  margin-bottom: 0;
  border-style: solid;
  border-bottom-width: 1px;
  border-color: @saGrey4;
}

.company {
  padding-bottom: @job-card-text-space;
  color: @saGrey1;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.greyLabel {
  padding: @job-card-text-space @job-card-label-indent;
  margin-right: @job-card-label-indent;
  background-color: @saGrey5;
  color: @saGrey2;
}

.classifiedLabel {
  &:extend(.greyLabel);
}

.confidentialLabel {
  &:extend(.greyLabel);
}

.featuredLabel {
  padding: @job-card-text-space @job-card-label-indent;
  margin-right: @job-card-label-indent;
  background-color: @panelPink;
  color: @saWhite;
}

.positionTitle {
  color: @linkBlue;
  padding-bottom: @job-card-text-space;
}

.sellingPointsList {
  list-style: disc;
  padding-left: @job-card-list-indent;
  color: @saGrey2;
}

.sellingPoint {
  padding-bottom: @job-card-text-space;
}

.jobDescriptionSection, .sellingPointsSection {
  padding-top: @job-card-section-space;
}

.bodyDescriptionText {
  color: @saGrey3; 
  padding-bottom: @job-card-text-space;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
}

.footerSection {
  padding-top: @job-card-section-space;
  display: flex;
  justify-content: space-between;
}

.footerLeft {
  flex-basis: auto;
  overflow: hidden;
}

.jobInfoList {    
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: space-between;
}

.jobInfo {
  padding-bottom: @job-card-text-space;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  color: @saGrey1;
  & span {
    vertical-align: middle;
  }
}

.postingDuration { 
  padding-bottom: @job-card-bottom-text-space;
  color: @saGrey3;
}

.companyLogoWrapper {
  width: 76px;
  min-width: 76px;
  height: 64px;
  margin-left: @job-card-logo-indent;
  border: 1px solid @saGrey4;
  display: flex;
  align-items: center;
  align-self: flex-end;
  overflow: hidden;
}

.companyLogo {
  max-width: 100%;
  max-height: 100%;
  margin: 0 auto;
}