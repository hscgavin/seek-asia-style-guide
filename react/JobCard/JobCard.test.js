import React from 'react';
import { shallow } from 'enzyme';

import JobCard from './JobCard';

const defaultJob = {
  company: 'SEEK Asia',
  jobTitle: 'Developer',
  location: 'Kuala Lumpur',
  salary: 'RM999999999 - RM999999999999',
  postingDuration: '1 hour ago'
};

describe('JobCard', () => {
  it('should render with default props', () => {
    const wrapper = shallow(<JobCard job={defaultJob} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with description props', () => {
    const descriptionJob = {
      ...defaultJob,
      description: 'Hello world'
    };

    const wrapper = shallow(<JobCard job={descriptionJob} />);
    expect(wrapper).toMatchSnapshot();
  });
});
