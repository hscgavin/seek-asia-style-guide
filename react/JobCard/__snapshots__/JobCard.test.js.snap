// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobCard should render with default props 1`] = `
<Card
  className="root"
>
  <Section>
    <Text
      baseline={true}
      className="company"
    >
      SEEK Asia
    </Text>
    <Text
      baseline={true}
      heading={true}
    >
      Developer
    </Text>
  </Section>
  <Section
    className="footer"
  >
    <div>
      <Text
        baseline={true}
      >
        <LocationIcon />
         
        Kuala Lumpur
      </Text>
      <Text
        baseline={true}
      >
        <MoneyIcon />
         
        RM999999999 - RM999999999999
      </Text>
      <Text
        baseline={true}
      >
        <TimeIcon />
         
        1h
      </Text>
    </div>
  </Section>
</Card>
`;

exports[`JobCard should render with description props 1`] = `
<Card
  className="root"
>
  <Section>
    <Text
      baseline={true}
      className="company"
    >
      SEEK Asia
    </Text>
    <Text
      baseline={true}
      heading={true}
    >
      Developer
    </Text>
  </Section>
  <Section
    className="bodyDescription"
  >
    <Text
      baseline={true}
    >
      Hello world
    </Text>
  </Section>
  <Section
    className="footer"
  >
    <div>
      <Text
        baseline={true}
      >
        <LocationIcon />
         
        Kuala Lumpur
      </Text>
      <Text
        baseline={true}
      >
        <MoneyIcon />
         
        RM999999999 - RM999999999999
      </Text>
      <Text
        baseline={true}
      >
        <TimeIcon />
         
        1h
      </Text>
    </div>
  </Section>
</Card>
`;
